Add migration guide to opt out of edge-to-edge by default on Android (#11068)
flutter/website: src/content/release/breaking-changes/default-systemuimode-edge-to-edge.md

## Context
Android is enforcing [edge-to-edge mode][2] by default for all apps targeting
Android 15 and above; see the [Android release notes][3] for more details on
this change. As a result, all Flutter apps running on Android that target
Android 15 and above will be opted into edge-to-edge mode by default, impacting
devices running on Android SDK 15+/API 35+ just like any other Android app.

At the time of publishing this guide, Flutter apps target Android 14 by
default and will not be opted into edge-to-edge mode automatically, but
your app will be impacted when you choose to target Android 15. If your app
targets `flutter.targetSdkVersion` (as it does by default), then your app will
target Android 15 starting with Flutter version 3.26, automatically opting your
app into edge-to-edge as a result; see the [timeline](#timeline) below for
details. If your app explicitly sets `SystemUiMode.edgeToEdge` for it to run in
edge-to-edge mode via a call to [`SystemChrome.setEnabledSystemUIMode`][4],
then your app is already migrated. Apps needing more time to migrate to
edge-to-edge mode will need to follow the steps below to opt out on
devices running Android SDK 15+.

Please note that 

> 1. Android plans for the workaround detailed below to be temporary.
> 2. Flutter plans to align with Android (and iOS) to
> support edge-to-edge by default within the year, so **please migrate to**
> **edge-to-edge before the operating system removes the ability to opt out**.

Reviewer: ```suggestion
1. Android plans for the workaround detailed here to be temporary.
2. Flutter plans to align with Android (and iOS) to
    support edge-to-edge by default within the year,
    so please **migrate to edge-to-edge mode before the operating system
    removes the ability to opt out**.
```

> ## Migration Guide
> To opt out of edge-to-edge by default on SDK 15, a new style attribute
> specification is required in each activity that needs an opt out. If
> you have a parent style that all other styles that need an opt out inherit
> from, then you can modify the parent only. In the example below, you will
> update the style generated from `flutter create`.

Reviewer: Our docs use sentence casing for titles and anchors.
We avoid using "above" or "below" in text. If it's translated to Chinese or Hebrew, for example,
above/below might incorrect. Another general writing rule.

```suggestion
## Migration guide

To opt out of edge-to-edge on SDK 15, specify the new style attribute
in each activity that requires it. If you have a parent style that child styles
need to opt out of, you can modify the parent only.
In the following example, update the style generated from `flutter create`.
```

> By default, the styles used in a Flutter app are set in the manifest
> (`your_app/android/app/src/main/AndroidManifest.xml`). Generally,
> styles used in your app are denoted by `@style` and help theme your app.
> Find these styles in your manifest. By default, you will find a normal theme
> that you need to migrate:

Reviewer: ```suggestion
By default, the styles used in a Flutter app are set in the manifest file
(`your_app/android/app/src/main/AndroidManifest.xml`). Generally,
styles are denoted by `@style` and help theme your app.
Modify these default styles in your manifest:
```

```xml
<manifest xmlns:android="http://schemas.android.com/apk/res/android">
    <application ...>
        <activity ...>
            <!-- Style you will need to modify: -->
            <meta-data
              android:name="io.flutter.embedding.android.NormalTheme"
              android:resource="@style/NormalTheme"
            />
        </activity>
    </application>
</manifest>
```

> To migrate this style, find where it is defined in
> `your_app/android/app/src/main/res/values/styles.xml`. There, add the
> following attribute to the style:

Reviewer: ```suggestion
Find where this style is defined in
`your_app/android/app/src/main/res/values/styles.xml`.
There, add the following attribute to the style:
```

```xml
<?xml version="1.0" encoding="utf-8"?>
<resources>
    ...
    <style name="NormalTheme" parent="@android:style/Theme.Light.NoTitleBar">
        ...
	    <!-- Add the following line: -->
        <item name="android:windowOptOutEdgeToEdgeEnforcement">true</item>
    </style>
</resources>
```

> This modified style will opt your app out of edge-to-edge by default for
> apps targeting Android SDK 15+, so now you are done!

Reviewer: ```suggestion
This modified style opts your app out of edge-to-edge for
apps targeting Android SDK 15+. So now you are done!
```
